src/azure-firewall/azext_firewall/custom.py

```diff
@@ -929,4 +972,81 @@ def remove_azure_firewall_policy_filter_rule(cmd, resource_group_name, firewall_
             target_rule_collection.rules.remove(rule)
     return client.create_or_update(resource_group_name, firewall_policy_name,
                                    rule_collection_group_name, rule_collection_group)
+
+
+def _judge(x, y):
+    return x if x else y
+
+
+# pylint: disable=too-many-locals
+def update_azure_firewall_policy_filter_rule(cmd, instance, rule_collection_name, rule_name,
+                                             description=None, ip_protocols=None, source_addresses=None,
+                                             destination_addresses=None, destination_ports=None,
+                                             protocols=None, fqdn_tags=None, target_fqdns=None,
+                                             source_ip_groups=None, destination_ip_groups=None,
+                                             translated_address=None, translated_port=None,
+                                             target_urls=None, enable_tls_inspection=None, web_categories=None):
+    (NetworkRule,
+     FirewallPolicyRuleApplicationProtocol,
+     ApplicationRule,
+     NatRule) = cmd.get_models('NetworkRule', 'FirewallPolicyRuleApplicationProtocol',
+                               'ApplicationRule', 'NatRule')
+    target_rule_collection = None
+    for rule_collection in instance.rule_collections:
+        if rule_collection.name == rule_collection_name:
+            target_rule_collection = rule_collection
+
+    if target_rule_collection is None:
+        raise CLIError("Cannot find corresponding rule, please check parameters")
```

Comment:
Please don't raise CLIError. You should use Errors defined in this [file](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli-core/azure/cli/core/azclierror.py)